docs/apis/calcite.md

```diff
@@ -0,0 +1,195 @@
+We will go over APIs from above code one by one. Class _**SQLExecEnvironment**_ provides a starting point and simple way to define metadata needed for running SQL statement. SQLExecEnvironment.getEnvironment() creates instance of SQLExecEnvironment.  
+
+    // Creates SQLExecEnvironment instance by using static method getEnvironment
+    SQLExecEnvironment sqlEnv = SQLExecEnvironment.getEnvironment();
+
+### Registering tables with SQLExecEnvironment
+Next, we need to register tables which can be used in a query. For this purpose, we can use _registerTable_ API from SQLExecEnvironment.
+
+    // Register KafkaEnpoint as table with kafka topic as table and specify data format as CSV
+    sqlEnv = sqlEnv.registerTable("ORDERS", new KafkaEndpoint("localhost:9090", 
+                                            "inputTopic", new CSVMessageFormat(<<csvSchema>>)));
+    
+    // Register FileEndpoint as tbale with HDFS file as table and specify data format as csv
+    sqlEnv = sqlEnv.registerTable("SALES", 
+                                          new FileEndpoint("/tmp/output", "out.file", 
+                                          new CSVMessageFormat(schemaOutDef)));
+
+    
+**_"registerTable"_** API takes the name of the table and data store endpoint as parameters. Endpoint signifies data storage mechanism and type of source/destination for data. Each of these endpoints requires a different type of configurations but all of them demand data format to be defined. In this example data format is defined using _**MessageFormat**_ by specifying it at endpoint creation. As of now 
+**_CSVMessageFormat_** is only supported **_Messageformat_** for _KafkaEndpoint_ and _FileEndpoint_. **_CSVMessageFormat_** can be configured with schema string as follows:
+
+    {"separator":",","quoteChar":"\"","fields":[{"name":"RowTime","type":"Date","constraints":{"format":"dd/MM/yyyy hh:mm:ss Z"}},{"name":"id","type":"Integer"},{"name":"Product","type":"String"},{"name":"units","type":"Integer"}]}
+
+Schema string is a JSON string defining a separator character for schema, quote character for fields with String types and fields along with their name, type and additional constraints if there are any.
+
+Currently, integration supports following data endpoints:
+
+* **KafkaEnpoint**
+: To define Kafka Endpoint we need to specify Kafka broker ( host:port ), topic name and MessageFormat as seen in line 1 in above code.
+* **FileEndpoint**
+: It needs to be configured with HDFS output folder, output file name and MessageFormat as in line 2 in above code 
+* **StreamEndpoint** 
+: This allows us to connect existing operator output or input to SQL query as a data source or sink respectively. StreamEndpoint needs immediate downstream operator's input port or immediate upstream operator's output port and field mapping for CSV data or POJO class. This will be explained in detail in next [example](http://192.168.1.31:8000/apis/calcite/#fustion-style-sql-application).
```

Comment:
Why is this pointing to 192.168.1.31? 
This IP won't be available to users of Apex.